tests/marketplace.spec.ts

```diff
@@ -254,19 +315,28 @@ describe('Marketplace tests', () => {
     await registerContract(deployer);
     const hash = string2ascii('h'.repeat(32));
 
-    const gas = (await marketplace.withSigner(bob).query.setNftContractHash(hash)).gasRequired;
-    const result = await marketplace.withSigner(bob).query.setNftContractHash(hash, {gasLimit: getEstimatedGas(gas)});
+    const gas = (await marketplace.withSigner(bob).query.setNftContractHash(NftContractType.rmrk, hash)).gasRequired;
+    const result = await marketplace.withSigner(bob).query.setNftContractHash(NftContractType.rmrk, hash, {gasLimit: getEstimatedGas(gas)});
 
     expect(result.value.unwrap().err.ownableError).to.equal('CallerIsNotOwner');
   });
 
-  it('factory works', async () => {
+  it('PSP34 factory works', async () => {
+    const shiden34Hash: Hash = shiden34.abi.info.source.wasmHash.toHex();
     await setup();
-    const marketplace_ipfs = 'ipfs://test';
-    const siden34Hash: Hash = shiden34.abi.info.source.wasmHash.toHex();
+    await callFactory(NftContractType.psp34, shiden34Hash);
+  });
 
-    const hashGas = (await marketplace.withSigner(deployer).query.setNftContractHash(siden34Hash)).gasRequired;
-    await marketplace.withSigner(deployer).tx.setNftContractHash(siden34Hash, { gasLimit: getEstimatedGas(hashGas) });
+  it('RMRK factory works', async () => {
+    const rmrkHash: Hash = rmrk.abi.info.source.wasmHash.toHex();
+    await setup();
+    await callFactory(NftContractType.rmrk, rmrkHash);
+  });
+
+  async function callFactory(contractType: NftContractType, contractHash: Hash) {
+    const marketplace_ipfs = 'ipfs://test';
+    const hashGas = (await marketplace.withSigner(deployer).query.setNftContractHash(contractType, contractHash)).gasRequired;
+    await marketplace.withSigner(deployer).tx.setNftContractHash(contractType, contractHash, { gasLimit: getEstimatedGas(hashGas) });
```

Comment:
in `getEstimatedGas` you should be ok without multiplying with 2

Reply:
Not multiplying by 2 works for most of the tests, except buy token test, which fails if I don't multiply gas.